WordPress/gutenberg
# `UnitControl`: fix controlled `unit` behavior

## Description


In https://github.com/WordPress/gutenberg/pull/37311#issuecomment-1054591492, @ntsekouras flagged a bug where the `UnitControl` component wouldn't correctly update its unit if the update was coming via the `value` prop (as opposed as from a user interaction).
For reference, these two patterns are also referred to as "controlled" or "uncontrolled" components. `UnitControl` should technically already support this behaviour, but the bug report shows otherwise.

This PR introduces a fix around that, alongside a new Storybook example and a unit test that mimic closely the scenario encountered by Nik when flagging the buggy behavior.

This PR also introduces a new dev dependency, [`@testing-library/user-event`](https://testing-library.com/docs/ecosystem-user-event/), a recent addition to Testing Library for handling interaction in unit tests better.

**Potential follow-ups:**

- Follow-up on https://github.com/WordPress/gutenberg/pull/37311, rewriting the code in a way that doesn't use the deprecated `unit` prop
- Mark the `unit` prop as deprecated also in TypeScript types, and consider adding a soft deprecation warning too
- Rewrite Storybook examples to use controls (and convert to TypeScript)
- Rewrite unit tests to avoid using classnames anduse the `user-event` library (and convert to TypeScript)

## Testing Instructions





- Run `UnitControl` unit tests, and make sure they pass
- Visit the "Controlled" Storybook example for `UnitControl` and make sure it works as expected (changing the unit in one control triggers that same unit change in the other control)

Comment out the fix added by this PR to the component:
- Run unit tests, the recently added test should fail
- Interact with the Storybook example, it should not work as expected

## Screenshots 

Before:

https://user-images.githubusercontent.com/1083581/156259323-075f7d23-1e58-465e-8b53-78759f5ea428.mp4

After:

https://user-images.githubusercontent.com/1083581/156259333-c970c778-0768-4049-aec5-c59d87dd142c.mp4



## Types of changes





## Checklist:
